PurchasesCoreSwift/Public/PurchaserInfo.swift

```diff
@@ -0,0 +1,317 @@
+     */
+    @objc public func purchaseDate(forEntitlement: String) -> Date? { entitlements[forEntitlement]?.latestPurchaseDate }
+
+    // TODO after migration make this internal and remove objc rename
+    @objc(JSONObject) public func jsonObject() -> [AnyHashable: Any] {
+        return originalData.merging(["schema_version": PurchaserInfo.currentSchemaVersion()], uniquingKeysWith: { (current, _) in current })
+    }
+
+    public override func isEqual(_ object: Any?) -> Bool {
+        guard let other = object as? PurchaserInfo else {
+            return false
+        }
+
+        var selfJson = self.jsonObject()
+        selfJson.removeValue(forKey: "request_date")
+        var otherJson = other.jsonObject()
+        otherJson.removeValue(forKey: "request_date")
+
+        return NSDictionary(dictionary: selfJson).isEqual(to: otherJson)
+    }
+
+    public override var description: String {
+        let activeSubsDescription = self.activeSubscriptions.reduce(into: [String: String]()) { dict, subId in
+            dict[subId] = "expiresDate: \(String(describing: self.expirationDate(forProductIdentifier: subId)))"
+        }
+
+        let activeEntitlementsDescription = self.entitlements.active.mapValues { $0.description }
+
+        let allEntitlementsDescription = self.entitlements.all.mapValues { $0.description }
+
+        var description = "<\(NSStringFromClass(type(of: self))): "
+        description += "originalApplicationVersion=\(String(describing: self.originalApplicationVersion)),\n"
+        description += "latestExpirationDate=\(String(describing: self.latestExpirationDate)),\n"
+        description += "activeEntitlements=\(activeEntitlementsDescription),\n"
+        description += "activeSubscriptions=\(activeSubsDescription),\n"
+        description += "nonSubscriptionTransactions=\(self.nonSubscriptionTransactions),\n"
+        description += "requestDate=\(String(describing: self.requestDate)),\n"
+        description += "firstSeen=\(String(describing: self.firstSeen)),\n"
+        description += "originalAppUserId=\(self.originalAppUserId),\n"
+        description += "entitlements=\(allEntitlementsDescription),\n"
+        description += ">"
+
+        return description
+    }
+
+    private struct SubscriberData {
+
+        let originalAppUserId: String
+        let managementURL: URL?
+        let originalApplicationVersion: String?
+        let originalPurchaseDate: Date?
+        let firstSeen: Date
+        let nonSubscriptions: [String: [[String: Any]]]
+        let entitlements: [String: Any]
+        let nonSubscriptionTransactions: [Transaction]
+        let allPurchases: [String: Any]
+        let expirationDatesByProduct: [String: Date?]
+        let purchaseDatesByProduct: [String: Date?]
+
+        init?(subscriberData: [String: Any]) {
+            let subscriptions = subscriberData["subscriptions"] as? [String: [String: Any]] ?? [String: [String: Any]]()
```

Comment:
I think you should be able to make the second part of this assignment `?? [:]`, since the compiler should have enough info to infer the type. 
same for others that do the same